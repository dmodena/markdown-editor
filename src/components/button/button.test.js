'use strict';

import React from 'react';
import Button from './index';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

const noop = () => null;

it('Should Button default to match snapshot', () => {
  const tree = renderer
    .create(<Button onClick={noop}>Click me default</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Should Button success to match snapshot', () => {
  const tree = renderer
    .create(
      <Button onClick={noop} kind="success">
        Click me success
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Should Button danger to match snapshot', () => {
  const tree = renderer
    .create(
      <Button onClick={noop} kind="danger">
        Click me danger
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button with prop kind = "success" should has class "-success"', () => {
  const wrapper = shallow(
    <Button onClick={noop} kind="success">
      Button Success
    </Button>
  );
  expect(wrapper.hasClass('-success')).toBe(true);
  expect(wrapper.hasClass('-danger')).toBe(false);
});

it('Button with prop kind = "success" should has class "-danger"', () => {
  const wrapper = shallow(
    <Button onClick={noop} kind="danger">
      Button danger
    </Button>
  );
  expect(wrapper.hasClass('-danger')).toBe(true);
  expect(wrapper.hasClass('-success')).toBe(false);
});
