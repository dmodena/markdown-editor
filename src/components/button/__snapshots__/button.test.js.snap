// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Button danger to match snapshot 1`] = `
<button
  className="button -danger"
  onClick={[Function]}
>
  Click me danger
</button>
`;

exports[`Should Button default to match snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Click me default
</button>
`;

exports[`Should Button success to match snapshot 1`] = `
<button
  className="button -success"
  onClick={[Function]}
>
  Click me success
</button>
`;
